src/main/java/com/greensopinion/gradle/android/eclipse/GenerateLibraryDependenciesAction.java

```diff
@@ -65,22 +65,23 @@ public void execute(Classpath classpath) {
 
 	private Stream<ClasspathEntry> explodeAarJarFiles(Library aarLibrary) {
 		File aarFile = new File(aarLibrary.getPath());
-		File targetFolder = new File(new File(new File(project.getProjectDir(), "build"), "exploded-aars"),
-				aarFile.getName());
+    String jarId = aarLibrary.getModuleVersion().toString().replaceAll(":", "-");
+    File targetFolder = new File(new File(new File(project.getProjectDir(), "build"), "exploded-aars"), jarId);
 		if (!targetFolder.exists()) {
 			if (!targetFolder.mkdirs()) {
 				throw new RuntimeException(format("Cannot create folder: {0}", targetFolder.getAbsolutePath()));
 			}
 			try (ZipFile zipFile = new ZipFile(aarFile)) {
-				zipFile.stream().forEach(e -> {
-					if (e.getName().endsWith(".jar")) {
-						File targetFile = new File(targetFolder, e.getName());
+				zipFile.stream().forEach(f -> {
+					if (f.getName().endsWith(".jar")) {
+            String targetName = jarId + ".jar";
+						File targetFile = new File(targetFolder, targetName);
 						ensureParentFolderExists(targetFile);
 						int index = 1;
 						while (targetFile.exists()) {
-							targetFile = new File(targetFolder, format("{0}_{1}", ++index, e.getName()));
+							targetFile = new File(targetFolder, format("{0}_{1}", ++index, targetName));
```

Comment:
Does incorporating an index into the filename still make sense, or could we go with the targetName alone?

Reply:
I guess the whole `while(exists)` part could be removed as it will copy each JAR to folders with unique names (and unique JAR names). Think "group-artifact-version" set is unique enough (at least Gradle uses this pattern).
If you prefer I could remove it and commit from my side.

Reply:
I'm prepared to accept this contribution as-is, since it looks good.  Given these are temporary files I don't think including the index is harmful, but certainly removing the while loop would simplify the code.  Feel free to push a separate pull request to remove the loop if you want, or we can just leave it.